api/setting.go

```diff
@@ -80,6 +81,52 @@ type (
 		MigrationSurchargeMultiplier uint64 `json:"migrationSurchargeMultiplier"`
 	}
 
+	// PricePinSettings contains the pinning settings used to pin certain
+	// settings to an underlying currency (e.g. USD). This uses an external
+	// explorer to retrieve the current exchange rate, allow users to configure
+	// certain prices in USD rather than SC.
+	PricePinSettings struct {
+		// Disabled should be set to true to disable price pinning. If enabled
+		// however, the currency must be set for the settings to be considered
+		// valid.
+		Disabled bool `json:"disabled"`
```

Comment:
I'll remove it and copy `hostd` where an empty `currency` means it's disabled. I personally don't like overloading fields with an added meaning, e.g. no currency means disabled. And then I made it negative because I wanted to enable by default (when set) and have users explicitly disable it when they want to. In the `bus` we've been defaulting our settings, that's why it's not really set or unset, the price pin settings are defaulted to a default value that disables the setting.